src/main/clojure/clara/rules/compiler.clj

```diff
@@ -179,10 +180,13 @@
         ;; Iterate through the bean properties, returning tuples and the corresponding methods.
         (for [^PropertyDescriptor property (seq (.. java.beans.Introspector
                                                     (getBeanInfo cls)
-                                                    (getPropertyDescriptors)))]
-
-          [(symbol (string/replace (.. property (getName)) #"_" "-")) ; Replace underscore with idiomatic dash.
-           (symbol (str "." (.. property (getReadMethod) (getName))))])))
+                                                    (getPropertyDescriptors)))
+              :let [read-method (.getReadMethod property)]
+              ;; In the event that there the class has an indexed property without a basic accessor we will simply skip
+              ;; the accessor as we will not know how to retrieve the value. see https://github.com/cerner/clara-rules/issues/446
+              :when read-method]
+          [(symbol (string/replace (.getName property) #"_" "-")) ; Replace underscore with idiomatic dash.
+           (symbol (str "." (.getName read-method)))])))
```

Comment:
You'll want to be careful that nothing here adds new reflection.  Perhaps compile a session (thus doing evals) with reflection warnings turned on.  This would be a good test to have actually though perhaps a bit out of scope here; as far as I know we don't have a test for this.

Reply:
Adding 
```
:global-vars {*warn-on-reflection* true}
```
to the project and building doesn't show this as a reflection.
To test that this is correct, i removed the type hint for PropertyDescriptor and i get the following:

> Reflection warning, clara/rules/compiler.clj:179:33 - reference to field getReadMethod on java.lang.Object can't be resolved.
Reflection warning, clara/rules/compiler.clj:183:36 - reference to field getName on java.lang.Object can't be resolved.
Reflection warning, clara/rules/compiler.clj:184:29 - reference to field getName can't be resolved.
Reflection warning, clara/rules/compiler.clj:179:33 - reference to field getReadMethod can't be resolved.
Reflection warning, clara/rules/compiler.clj:183:36 - reference to field getName can't be resolved.
Reflection warning, clara/rules/compiler.clj:184:29 - reference to field getName can't be resolved.

So, i think we are good for this example.

Reply:
Yep, doesn't look like reflection.

I think we should just have
```clj
:global-vars {*warn-on-reflection* true}
```
at the top-level of our lein project.clj though in general

Reply:
why not:
https://github.com/cerner/clara-rules/pull/449/commits/dc754f90f170c0ff6b95133443bda1d361a91463

Reply:
To be clear, my concern wasn't necessarily with reflection warnings in Clara itself but rather with reflection when invoking accessors on Java objects.  For example:

```
(defrule testrule
    [JavaBeanWithFieldA (= a "some value")]
   =>
  ;; do something
)
```
reflecting when invoking the .getA method behind the scenes.  However in any case if that were happening I'd expect to see more warnings in the Travis build with global reflection warnings enabled - in particular, warnings interspersed through test execution when eval calls build rule networks.  So +1.

Reply:
Ah, yeah that makes sense. I believe the reason that the rules network fns don't experience reflection is because we add the metadata a bit later:
https://github.com/cerner/clara-rules/blob/1bafa05c2502eb452bdd765c1b45c158725f575e/src/main/clojure/clara/rules/compiler.clj#L390